### ISSUE TITLE
Connection String environment variable doesn't work in containers

### ISSUE THREAD

**Issue by robbie-coder** [microsoft/ApplicationInsights-Java#1943]
### Expected behavior
When setting the APPLICATIONINSIGHTS_CONNECTION_STRING environment variable in a Windows or Linux container I expect the java agent to pick this up and use it. I orginally encountered this in Windows container but I can also reproduce it in a Linux container.

### Actual behavior
The applicationinsights.log file logs: No connection string or instrumentation key provided. While the java program itself can correctly read the environment variable.

### To Reproduce
Prerequist: have docker for windows running on your machine, see further below for linux instructions.
1. Open the attached [example.zip](https://github.com/microsoft/ApplicationInsights-Java/files/7454246/example.zip) and extract it to a local folder
2. Open the Dockerfile and change the 0000 id in the ENV variable line to a real AI id, save the file
3. Do a docker build to create the image
4. Run the image with docker run

When I change the FROM line in the Dockerfile to _FROM openjdk:11_ I can also reproduce this in a Linux container. 

The java program will output the environment variable and you will see that it can correctly read it. You will also see the output of the javagent which reports that no connection string was set.
<pre>
2021-11-01 16:16:39.695+01:00 ERROR c.m.applicationinsights.agent -
*************************
ApplicationInsights Java Agent 3.2.2 failed to start
*************************

Description:
No connection string or instrumentation key provided

Action:
Please provide connection string or instrumentation key.

Outputting the APPLICATION_INSIGHTS_CONNECTION_STRING environment variable:
InstrumentationKey=00000000-0000-0000-0000-000000000000;IngestionEndpoint=https://westeurope-5.in.applicationinsights.azure.com/
</pre>

### System information
Please provide the following information:
 - Agent Version: 3.2.2
 - OS type and version: Windows servercore:ltsc2019 container image
 - Using spring-boot? no

### Logs
<pre>2021-11-01 15:24:30.227+01:00 ERROR c.m.applicationinsights.agent - 
*************************
ApplicationInsights Java Agent 3.2.2 failed to start
*************************

Description:
No connection string or instrumentation key provided

Action:
Please provide connection string or instrumentation key.
</pre>

### Additional information
In the example I set the environment variable in the Dockerfile while in my real application it is set by the docker-compose.yml file but the behavior in both cases is identical.

When I put the exact same connection string in the json config file it works correctly, so I know that string itself is correct.

If I test this with an environment variable on a Windows Server 2019 machine (not a container) it does work correctly with the same Java program. I can only reproduce this in containers.

Let me know if I can perform any additional tests or information.

**Comment by trask** [comment]
> ### Expected behavior
> When setting the APPLICATIONINSIGHTS_CONNECTION_STRING environment variable in a Windows or Linux container I expect the java agent to pick this up and use it. I orginally encountered this in Windows container but I can also reproduce it in a Linux container.

hi @robbie-coder! it looks like you have an extra underscore `_` in your test:

`APPLICATION_INSIGHTS_CONNECTION_STRING` instead of `APPLICATIONINSIGHTS_CONNECTION_STRING`

**Comment by robbie-coder** [comment]
Thanks for your feedback @trask in this example application I indeed made a typo. After fixing the typo I could no longer reproduce it in the example. However in my real application I did not make that typo and still had the problem.

But this got me thinking why is this only occurring in a container and only with my application not with a simple application. I eventually found the real cause.

On Windows environment variables can be set on three different levels:
- Process
- User
- Machine

If you set an env var through docker it is set on the Process level. In my case the actual CMD in the Dockerfile would start a PowerShells script. That script would do some stuff and in the end start a Windows Service. That Windows Service is my actual Java app that loads that javaagent.

Now that Windows Service is registered under a different user then the user that the PowerShell script is running that is started by Docker. That explains why the Java app could not see the env var that docker had set, because it was for a different user and process.

To fix this I simply promote the user level connection string env var to the Machine level before I start the Windows Service:
`if ($Env:APPLICATIONINSIGHTS_CONNECTION_STRING) {
    [System.Environment]::SetEnvironmentVariable("APPLICATIONINSIGHTS_CONNECTION_STRING", $Env:APPLICATIONINSIGHTS_CONNECTION_STRING, 'Machine')
}`


After that everything works fine. So it was all caused by configuration error on my side and a bit of ignorance on how the different levels of env vars work on Docker and Windows.

Thanks for looking at this I'll close the issue as the java agent itself is working fine.
